### ISSUE TITLE
bug(dialog): MDC dialog text color fails to pass WCAG AAA contrast tests

### ISSUE THREAD

**Issue by PowerKiKi** [angular/components#26797]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

14

### Description

The MDC dialog content has a poor contrast that fails to pass WCAG AAA tests, as seen in screenshot.

In both cases the background is white, `#ffffff`. Before, in v14, the text was `#212121` with a contrast of 16.1:1, according to https://webaim.org/resources/contrastchecker/. After, in v15 with MDC, the new text color is `#666666`, which clearly has a lower contrast of only 5.74:1.

Screenshot straight from https://v14.material.angular.io/components/dialog/overview and https://rc.material.angular.io/components/dialog/overview: 

![image](https://user-images.githubusercontent.com/72603/225563921-0e0722ea-0e76-49cf-be4f-8d5753eda27f.png)

![image](https://user-images.githubusercontent.com/72603/225564505-387ab2a2-50cc-4629-a0ff-1e5ee3e0c8eb.png)

### Reproduction

See screenshots

### Expected Behavior

For such as simple and common use-case, I expect it to pass even the strictest contrast tests, such as WCAG AAA.

### Actual Behavior

It fails WCAG AAA, and thus is less readable in sub-optimal conditions.

### Environment

- Angular: 15
- CDK/Material: 15
- Browser(s): Chrome 111
- Operating System (e.g. Windows, macOS, Ubuntu): Ubuntu 22.04

**Comment by PowerKiKi** [comment]
# Brittle workaround

We did found a workaround with the code below. However I don't think overriding CSS value is the proper way to solve this issue. 

```scss
@mixin my-mixin($theme) {

    // Since MDC, dialog text color is too pale, so we restore a higher contrast for better readability
    .mat-mdc-dialog-container .mdc-dialog__content {
        color: mat.get-color-from-palette(map.get($theme, foreground), base, 0.87);
    }
}
```

In general the theming API is very hard to work with. The documentation lives only in source code, and it is often quite terse, specifically it lacks usage examples. This is especially disappointing when compared to the awesome doc of TypeScript that is available on the angular web site, searchable, and detailed with usage examples.


# The lib breaks the spec

[The spec](https://m3.material.io/components/dialogs/specs), says that the background color of the container should be `md.sys.color.surface-container-high` and the color of supporting text should be `md.sys.color.on-surface-variant`:

![image](https://user-images.githubusercontent.com/72603/236116962-c9fbaea5-6e96-4660-b341-93043b60294a.png)


But the library definitely does not use `md.sys.color.surface-container-high` , because the background is not purple-ish but white. Instead it seems to be using `md.sys.color.surface-container-lowest`. So the library is already breaking the specs. (and thanks for that, I am not a fan of colored background dialogs)

![image](https://user-images.githubusercontent.com/72603/236117191-40290ec0-5a9e-4beb-a9ff-c1e47032630b.png)


So if the background color is `surface`, then according to [the theme builder](https://m3.material.io/theme-builder#/custom), then the fore color should be `#1C1B1F`, and that would indeed pass WCAG AAA:

![image](https://user-images.githubusercontent.com/72603/236118405-bd69364d-408e-4ca7-9590-af6a878496d7.png)

# Conclusion

@zarend, this issue should be reopened (but I cannot), until either:

- it's closed as won't fix
- a proper way to customize the dialog background and text color with Angular theming API is shared
- the dialog text contrast pass WCAG AAA (as the spec does)

**Comment by yogeshgadge** [comment]
I am stealing your brittle workaround. 
Not sure what the MDC/MAT decided to pale dialog's  content.
